addons/arsenal/functions/fnc_fillRightPanel.sqf

```diff
@@ -42,101 +42,111 @@ if (!(ctrlShown _searchbarCtrl) || {ctrlFade _searchbarCtrl > 0}) then {
     };
 
-    _cachedItemInfo params ["_displayName","_picture", "_mass"];
+    _cachedItemInfo params ["_displayName", "_picture", "_mass"];
 
     private _lbAdd = _ctrlPanel lnbAddRow ["", _displayName, "0"];
-    private _columns = count lnbGetColumnsPosition _ctrlPanel;
 
-    _ctrlPanel lnbSetData [[_lbAdd, 0], _x];
+    _ctrlPanel lnbSetData [[_lbAdd, 0], _className];
     _ctrlPanel lnbSetPicture [[_lbAdd, 0], _picture];
     _ctrlPanel lnbSetValue [[_lbAdd, 0], _mass];
-    _ctrlPanel setVariable [_x, _mass];
-    _ctrlPanel lnbSetValue [[_lbAdd, 2], [0, 1] select (_isUnique)];
-    _ctrlPanel lbSetTooltip [_lbAdd * _columns,format ["%1\n%2", _displayName, _x]];
+    _ctrlPanel lnbSetValue [[_lbAdd, 2], [0, 1] select _isUnique];
+    _ctrlPanel lbSetTooltip [_lbAdd * (count lnbGetColumnsPosition _ctrlPanel), format ["%1\n%2", _displayName, _className]];
+    _ctrlPanel setVariable [_className, _mass];
 };
 
-// Retrieve compatible mags
-private _compatibleItems = [];
-private _compatibleMagazines = [[[], []], [[], []], [[], []]];
-{
-    if (_x != "") then {
-        private _weaponConfig = (configFile >> "CfgWeapons" >> _x);
-        private _index = _forEachIndex;
-
-        {
-            private _subIndex = _forEachIndex min 1;
-            {
-                ((_compatibleMagazines select _index) select _subIndex) pushBackUnique (configName (configFile >> "CfgMagazines" >> _x))
-            } foreach ([getArray (_weaponConfig >> _x >> "magazines"), getArray (_weaponConfig >> "magazines")] select (_x == "this"));
+// Get all weapons on unit
+private _weapons = [primaryWeapon GVAR(center), handgunWeapon GVAR(center), secondaryWeapon GVAR(center), binocular GVAR(center)];
+private _compatibleMagazines = [];
+private _muzzles = [];
 
-            // Magazine groups
-            {
-                private _magazineGroups = uiNamespace getVariable [QGVAR(magazineGroups), createHashMap];
-                private _magArray = _magazineGroups get (toLower _x);
-                {((_compatibleMagazines select _index) select _subIndex) pushBackUnique _x} forEach _magArray;
-            } foreach ([getArray (_weaponConfig >> _x >> "magazineWell"), getArray (_weaponConfig >> "magazineWell")] select (_x == "this"));
+// Get all compatibles magazines with unit's weapons
+{
 
+    _muzzles = _x call CBA_fnc_getMuzzles;
 
-        } foreach getArray (_weaponConfig >> "muzzles");
-    };
-} foreach [primaryWeapon GVAR(center), handgunWeapon GVAR(center), secondaryWeapon GVAR(center)];
-
-private _itemsToCheck = [];
-private _compatibleMagsPrimaryMuzzle = [];
-private _compatibleMagsSecondaryMuzzle = [];
+    // Get compatible magazines for primary & secondary muzzle (secondary muzzle is not guaranteed to exist)
+    // Assumption: One weapon can have two muzzles maximum
+    _compatibleMagazines pushBack [compatibleMagazines [_x, _muzzles param [0, ""]], compatibleMagazines [_x, _muzzles param [1, ""]]];
```

Comment:
Can be easily swapped out for `CBA_fnc_compatibleMagazines`

Reply:
Decision has been taken to move forward with engine commands instead of CBA. If bugs are found with engine commands, ACE will revert back to the CBA functions.